test/node-report/test-exception.js

```diff
@@ -1,42 +1,24 @@
+// Flags: --experimental-report --diagnostic-report-uncaught-exception
 'use strict';
-
-// Testcase to produce report on uncaught exception
+// Test producing a report on uncaught exception.
 const common = require('../common');
 common.skipIfReportDisabled();
-if (process.argv[2] === 'child') {
-  function myException(request, response) {
-    const m = '*** test-exception.js: throwing uncaught Error';
-    throw new Error(m);
-  }
+const assert = require('assert');
+const helper = require('../common/report');
+const tmpdir = require('../common/tmpdir');
+const error = new Error('test error');
 
-  myException();
+common.expectWarning('ExperimentalWarning',
+                     'report is an experimental feature. This feature could ' +
+                     'change at any time');
+tmpdir.refresh();
+process.report.setOptions({ path: tmpdir.path });
 
-} else {
-  const helper = require('../common/report.js');
-  const tmpdir = require('../common/tmpdir');
-  tmpdir.refresh();
-  const spawn = require('child_process').spawn;
-  const assert = require('assert');
+process.on('uncaughtException', common.mustCall((err) => {
+  assert.strictEqual(err, error);
+  const reports = helper.findReports(process.pid, tmpdir.path);
+  assert.strictEqual(reports.length, 1);
+  helper.validate(reports[0]);
+}));
 
-  const child = spawn(process.execPath,
-                      ['--experimental-report',
-                       '--diagnostic-report-uncaught-exception',
-                       __filename, 'child'],
-                      { cwd: tmpdir.path });
-  // Capture stderr output from the child process
-  let stderr = '';
-  child.stderr.on('data', (chunk) => {
-    stderr += chunk;
-  });
-  child.on('exit', common.mustCall((code) => {
-    const report_msg = 'No reports found';
-    const process_msg = 'Process exited unexpectedly';
-    assert.strictEqual(code, 1, process_msg + ':' + code);
-    assert.ok(new RegExp('myException').test(stderr),
-              'Check for expected stack trace frame in stderr');
```

Comment:
The test no longer tests for the stack frame in the report?

Reply:
Prior to this change, a child process would throw an exception, and the test verified that the error was part of stderr. With this change, an exception is thrown, and the error is tested using `assert.strictEqual()`. The contents of the report were never really checked. Improving the report validation across all of the tests is on my todo list.